src/functions.php

```diff
@@ -19,3 +19,15 @@ function getDataArray(object $obj): array
 {
     return json_decode(json_encode($obj), true);
 }
+
+/**
+ * @param int $seconds
+ * @return DateInterval
+ * @throws Exception
+ */
+function getDateInterval(int $seconds): DateInterval
+{
+    $day1 = date_create('@0');
+    $day2 = date_create('@0')->add(new DateInterval('PT' . $seconds . 'S'));
+    return $day2->diff($day1, true);
```

Comment:
Why not `return new DateInterval('PT' . $seconds . 'S')`?

Reply:
Because it's different things. I need this function if i want to get pure date as '1 month' or '1 year'. 2592000 seconds just gives us 30 days and doesn't matter what month it is.

Reply:
`        $time1 = new \DateTime('2021-02-01');
        $time2 = new \DateTime('2021-02-01');
        $interval1 = new \DateInterval('PT2592000S');
        $interval2 = new \DateInterval('P1M');
        var_dump($time1->add($interval1));
        var_dump($time2->add($interval2));
`

Reply:
`
object(DateTime)#351 (3) {
  ["date"]=>
  string(26) "2021-03-03 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(13) "Europe/Berlin"
}
object(DateTime)#354 (3) {
  ["date"]=>
  string(26) "2021-03-01 00:00:00.000000"
  ["timezone_type"]=>
  int(3)
  ["timezone"]=>
  string(13) "Europe/Berlin"
}
`

Reply:
A good point!
When we speak about intervals they are always depend on the date. In real world you never buy a subscription for 2592000 seconds, you subscribe for one month.

You've made unix timestamp 0 starting point for your calculations which is the 1st of January 1970. I see no reason to use this day as a reference.

What do you think if we store the periods as string, in the syntax that DateInterval constructor use?


P.S. read this interesting article about common mistakes programmers do about the time https://infiniteundo.com/post/25509354022/more-falsehoods-programmers-believe-about-time

Reply:
Thx for the interesting article. I never assume what so many unbelievable moments about 'time'. I don't believe what all this assumptions are false still)
I have change time storage format to Dateinterval sintaxes